sysctl: use range for user.max_fanotify_marks check

ddiss opened the pull request:
The user.max_fanotify_marks sysctl is set with a formula that takes addressable memory into account. Instead of assuming memory resources of the test system, just confirm that it sits within the min=FANOTIFY_OLD_DEFAULT_MAX_MARKS, max=FANOTIFY_DEFAULT_MAX_USER_MARKS bounds.

https://bugzilla.suse.com/show_bug.cgi?id=1218234

ddiss commented:
The bz link in the commit description is wrong - updating...

foursixnine reviewed (approved):
Now it makes much more sense (I had the comment pending but never submitted it) 

@ddiss, one thing: I noticed you changed the value of the upper limit to a lower value from your previous commit; why is that?

@josegomezr when we were adding that threshold, and talked about exactly this, You were mentioning 0.5 was too small right? can I get your two cents?

ddiss commented:
@foursixnine thanks for the feedback...

> Now it makes much more sense (I had the comment pending but never submitted it)
> 
> @ddiss, one thing: I noticed you changed the value of the upper limit to a lower value from your previous commit; why is that?

Not sure I follow - the range that I've changed it to directly matches the kernel min/max logic:
```
  30 #define FANOTIFY_DEFAULT_MAX_EVENTS     16384
  31 #define FANOTIFY_OLD_DEFAULT_MAX_MARKS  8192
  32 #define FANOTIFY_DEFAULT_MAX_GROUPS     128
...          
  42 #define FANOTIFY_DEFAULT_MAX_USER_MARKS \
  43         (FANOTIFY_OLD_DEFAULT_MAX_MARKS * FANOTIFY_DEFAULT_MAX_GROUPS)
...
2010         max_marks = clamp(max_marks, FANOTIFY_OLD_DEFAULT_MAX_MARKS,
2011                                      FANOTIFY_DEFAULT_MAX_USER_MARKS);
```

I've added one follow up commit to change the inotify check to also use a range, given that it's also based on addressable memory.

josegomezr commented:
@foursixnine the logic I placed in #9 relies on the fact that the machines have a "fixed-enough" RAM size by default (IIRC 8G, but that G may drift a bit), and under that condition the `%0.5` works fine.

but as @ddiss asserts, following the Kernel logic will allow scheduling the same test in theoretically any amount of RAM without needing to alter the RAM.

I got to the same issue with numbers varying slightly when investigating it, see: https://progress.opensuse.org/issues/135722#note-6

foursixnine commented:
We're good now, we might later change this again, waiting on a conversation with some folks during Labs conference, for now  comments in https://bugzilla.suse.com/show_bug.cgi?id=1219366 are sufficient

rfan1 commented:
Hello folks, thanks for taking care of https://bugzilla.suse.com/show_bug.cgi?id=1219366

I fixed some format issue then https://github.com/openSUSE/sys-param-check/pull/13
